### PR TITLE
Schema name properties xml jdbc

### PR DESCRIPTION
**Pull request by zjcnb**
For #11621 

properties and namespace xml add schema name

### CODE REVIEW

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+package org.apache.shardingsphere.spring.namespace.tag;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * schema name bean definition tag.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class SchemaNameBeanDefinitionTag {
```
Comment (taojintianxia) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java:27]: is this a final class ?

File: shardingsphere-test/shardingsphere-integration-scaling-test/shardingsphere-integration-scaling-test-mysql/src/test/java/org/apache/shardingsphere/integration/scaling/test/mysql/env/config/SourceConfiguration.java [hunk 2]
```diff
@@ -94,6 +94,6 @@ private static YamlRootRuleConfigurations getShardingJdbcConfiguration(final Str
     public static DataSource createHostDataSource(final Map<String, YamlTableRuleConfiguration> tableRules) {
         ShardingSphereJDBCDataSourceConfiguration configuration = getHostConfiguration(tableRules);
         return new ShardingSphereDataSource(new YamlDataSourceConfigurationSwapper().swapToDataSources(configuration.getRootRuleConfigs().getDataSources()),
-                new YamlRuleConfigurationSwapperEngine().swapToRuleConfigurations(configuration.getRootRuleConfigs().getRules()), null);
+                new YamlRuleConfigurationSwapperEngine().swapToRuleConfigurations(configuration.getRootRuleConfigs().getRules()), null, null);
```
Comment (taojintianxia) [inline shardingsphere-test/shardingsphere-integration-scaling-test/shardingsphere-integration-scaling-test-mysql/src/test/java/org/apache/shardingsphere/integration/scaling/test/mysql/env/config/SourceConfiguration.java:97]: why pass null as the parameter ?

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.spring.namespace.tag;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * schema name bean definition tag.
```
Comment (strongduanmu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java:24]: @zjcnb Please keep the first letter capitalized.

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+package org.apache.shardingsphere.spring.boot.schema;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.StandardEnvironment;
+
+/**
+ * schema name setter.
```
Comment (strongduanmu) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-spring-infra/shardingsphere-jdbc-spring-boot-starter-infra/src/main/java/org/apache/shardingsphere/spring/boot/schema/SchemaNameSetter.java:26]: @zjcnb Same problem.

File: shardingsphere-test/shardingsphere-integration-scaling-test/shardingsphere-integration-scaling-test-mysql/src/test/java/org/apache/shardingsphere/integration/scaling/test/mysql/env/config/SourceConfiguration.java (hunk 2 above)
Comment (zjcnb) [reply shardingsphere-test/shardingsphere-integration-scaling-test/shardingsphere-integration-scaling-test-mysql/src/test/java/org/apache/shardingsphere/integration/scaling/test/mysql/env/config/SourceConfiguration.java:97]: The parameter comes from the user configuration, i don't sure real value. By the way, if it is a null value, I will assign it a default value.

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java (hunk 1 above)
Comment (zjcnb) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java]: yes, i fixed it

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java (hunk 3 above)
Comment (zjcnb) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/main/java/org/apache/shardingsphere/spring/namespace/tag/SchemaNameBeanDefinitionTag.java]: okay

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**